base/collect_database.go

```diff
@@ -0,0 +1,315 @@
+
+	// calculate the ElapsedTime based on the StartingTime if it has been provided
+	currentTime := time.Now().Unix()
+	return currentTime - startingTime, nil
+}
+
+// construct request to database and return extracted metric value
+//
+// bool return value represents whether the pipeline was able to run to
+// completion (prevents double error statement)
+func queryDatabaseAndGetValue(template CollectDatabaseTemplate, metric Metric) (interface{}, bool) {
+	// create a new HTTP request
+	req, err := http.NewRequest(template.Method, template.Url, nil)
+	if err != nil {
+		log.Logger.Error("could not create new request for metric ", metric.Name, ": ", err)
+		return nil, false
+	}
+
+	// iterate through headers
+	for headerName, headerValue := range template.Headers {
+		req.Header.Add(headerName, headerValue)
+	}
+	req.Header.Add("Content-Type", "application/json;charset=utf-8")
+
+	// add query params
+	q := req.URL.Query()
+	params := metric.Params
+	for _, param := range params {
+		q.Add(param.Name, param.Value)
+	}
+	req.URL.RawQuery = q.Encode()
+
+	// send request
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Logger.Error("could not request metric ", metric.Name, ": ", err)
+		return nil, false
+	}
+	defer resp.Body.Close()
+
+	// read response body
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Logger.Error("could not read response body for metric ", metric.Name, ": ", err)
+		return nil, false
+	}
+
+	// JSON parse response body
+	var jsonBody interface{}
+	err = json.Unmarshal([]byte(body), &jsonBody)
+	if err != nil {
+		log.Logger.Error("could not JSON parse response body for metric ", metric.Name, ": ", err)
+		return nil, false
+	}
+
+	// perform jq expression
+	query, err := gojq.Parse(metric.JqExpression)
+	if err != nil {
+		log.Logger.Error("could not parse jq expression \""+metric.JqExpression+"\" for metric ", metric.Name, ": ", err)
+		return nil, false
+	}
+	iter := query.Run(jsonBody)
+
+	value, ok := iter.Next()
+	if !ok {
+		log.Logger.Error("could not extract value with jq expression for metric ", metric.Name, ": ", err)
+		return nil, false
+	}
+
+	return value, true
+}
+
+// run executes this task
+func (t *collectDatabaseTask) Run(exp *Experiment) error {
+	// validate inputs
+	var err error
+
+	err = t.validateInputs()
+	if err != nil {
+		return err
+	}
+
+	// initialize defaults
+	t.initializeDefaults()
+
+	// get current directory path
+	path, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// collect all files paths in current directory that ends with metrics.yaml
+	metricFilePaths := []string{}
+	err = filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !info.IsDir() && strings.HasSuffix(path, "metrics.yaml") {
+			metricFilePaths = append(metricFilePaths, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// collect metrics for all metric files and versionInfos
```

Comment:
Before you enter this for loop, you may need to initialize results.

Please see https://github.com/iter8-tools/iter8/blob/800fe74b59559e99aeeb55ad4d2dc060472161a2/base/collect_grpc.go#L241-L244 for example.

Reply:
Done!